src/cycax/cycad/engines/cycax_part_freecad.py

```diff
@@ -224,43 +225,73 @@ def hole(
             cyl.Placement = App.Placement(Vector(x, y, z), App.Rotation(Vector(0, 1, 0), 270))
         return cyl
 
-    def render_to_png(self, target_path: Path):
-        """This method will be used for creating a png of 2 specific views of the object.
+
+    def render_to_png(self, view: str|None = None):
+        """Used to create a png of the desired side.
+        
         Args:
-            target_path: The Path the png needs to be saved under.
+            view: The side of the object the png will be produced from.
+            
         """
         active_doc = FreeCADGui.activeDocument()
-        target_image_file = f"{target_path}-perspectiveAll.png"
-        FreeCADGui.SendMsgToActiveView("ViewFit")
-        active_doc.activeView().viewAxometric()
-        active_doc.activeView().fitAll()
-        active_doc.activeView().saveImage(str(target_image_file), 2000, 1800, "White")
-        target_image_file = f"{target_path}-perspectiveTop.png"
+        view = view.upper()
+
+        if view is None:
+            view = "ALL"
```

Comment:
Move this to change_view. Make change view more versatile and give it an optional default attribute like `get` on `dict`.
method signature:
```
 def change_view(self, active_doc, side: str, default: str=None):

if side is None:
    side = default
if side is None:
    raise Some Error

...
```

Reply:
second if side is None is not needed and since it is followed by a match. Doing nothing for no side might be ok.